docs/design/2024-09-06-maintain-priority-queue-in-memory.md

```diff
@@ -0,0 +1,556 @@
+
+## Current Implementation
+
+### Calculate the score
+
+In the current implementation, we use a weighted scoring approach to maintain the queue, which is our current scoring criteria.
+
+Weight table:
+
+| Name                 | Meaning                                                                                                                                                                                       | Weight                                                                                                                                                                                                                                                                                                              | Data Source                  |
+| -------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------- |
+| Percentage of Change | The percentage of change since the last analysis.                                                                                                                                             | log10(1 + Change Ratio). [Check the graph](https://www.desmos.com/calculator/gb3i2cropz). <br><br> Note: For unanalyzed tables, we set the percentage of changes to 100%                                                                                                                                            | Stats Cache                  |
+| Table Size           | The size is equal to the number of rows multiplied by the number of columns in the table that are subject to analysis. The smaller tables should have a higher priority than the bigger ones. | Applying a logarithmic transformation, namely log10(1 + Table Size), and its 'penalty' is calculated as 1 - log10(1 + Table Size). [Check the graph.](https://www.desmos.com/calculator/x3fq76w8sb)                                                                                                                 | Stats Cache + Table Info     |
+| Analysis Interval    | Time since the last analysis execution for the table. The bigger interval should have a higher priority than the smaller interval.                                                            | Applying a logarithmic transformation, namely log10(1 + Analysis Interval). To further compress the rate of growth for larger values, we can consider taking the logarithmic square root of x. The final formula is log10(1 + √Analysis Interval). [Check the graph.](https://www.desmos.com/calculator/plhtkfqhx9) | Stats Cache                  |
```

Comment:
what's the unit of interval? If Analysis Interval >> 1, "log10(1 + √Analysis Interval)" can be approximately equal to "0.5 * log10(Analysis Interval)", it does not compress the rate of growth for larger values well.

Reply:
The unit is second.